components/dashboard/src/service/service.ts

```diff
@@ -38,18 +47,23 @@ function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
                 }
             }
         });
+
+        if (_websocket && "reconnect" in _websocket) {
+            reconnect = () => { (_websocket as any).reconnect() };
+        }
     }
     const service = new GitpodServiceImpl<C, S>(proxy);
+    (service as any).reconnect = reconnect;
     return service;
 }
 
-
-let gitpodService: GitpodService;
-
-const reconnect = () => {
-    gitpodService = createGitpodService();
+declare global {
+    type GitpodService = ReturnType<typeof createGitpodService> & { reconnect: () => void }
+    interface Window { gitpodService?: GitpodService; }
 }
 
-reconnect();
+// reuse existing service object if present
+let gitpodService: GitpodService = window.gitpodService || (window.gitpodService = createGitpodService() as GitpodService);
```

Comment:
we should only reference the global here, so there's no need for types. Also we should probably create a namespace object. something like `window._gp.gitpodService`.